php/Blockonomics.php

```diff
@@ -238,4 +294,334 @@ public function testSetup()
+        $this->load_blockonomics_header($template_name);
+
+        // Load the selected template
+        // Check if child theme or parent theme have overridden the template
+        $template = 'blockonomics_'.$template_name.'.php';
+        if ( $overridden_template = locate_template( $template ) ) {
+            load_template( $overridden_template );
+        } else {
+            load_template( plugin_dir_path(__FILE__)."../templates/" .$template );
+        }
+
+        $this->load_blockonomics_footer($template_name);
+
+        exit();
+    }
+
+    // Load the the payment confirmed template in the page
+    public function load_nojs_payment_confirmation_template(){
+        $this->load_blockonomics_template('nojs_payment_confirmation');
+    }
+
+    // Check if any pending payments linked to the order
+    public function check_for_pending_payment($orders, $order_id){
+        $network_confirmations = get_option("blockonomics_network_confirmation",2);
+        foreach ($orders as $address => $order){
+            if ($order['status'] >= 0 && $order['status'] < $network_confirmations){
+                $this->redirect_finish_order($show_order);
+            };
+        };
+    }
+
+    public function update_waiting_order($order){
+        $wc_order = new WC_Order($order['order_id']);
+        $order['value'] = $wc_order->get_total();
+        $order['currency'] = get_woocommerce_currency();
+        if(get_woocommerce_currency() != 'BTC'){
+            $responseObj = $this->get_price($order['currency'], $order['crypto']);
+            if($responseObj->response_code != 200) {
+                exit();
+            }
+            $price = $responseObj->price;
+            $price = $price * 100/(100+get_option('blockonomics_margin', 0));
+        }else{
+            $price = 1;
+        }
+        $order['satoshi'] = intval(round(1.0e8*$wc_order->get_total()/$price));
+        $order['timestamp'] = time();
+        return $order;
+    }
+
+    // Get any existing unpaid order for the crypto
+    public function get_waiting_order($orders, $order_id, $crypto){
+        foreach ($orders as $address => $order) {
+            if($order['crypto'] && $crypto) {
+                // Check if order and has expired
+                if ( $order['timestamp'] <= time() - get_option("blockonomics_timeperiod") * 60 ) {
+                    $order = $this->update_waiting_order($order);
+                }
+                return $order;
+            }
+        }
+        return false;
+    }
+
+    // Save the new address to the WooCommerce order
+    public function record_address($order_id, $crypto, $address){
+        update_post_meta($order_id, $crypto .'_address', $address);
+    }
+
+    public function create_new_order($order_id, $crypto){
+        $responseObj = $this->new_address(get_option("blockonomics_callback_secret"), $crypto);
+        if($responseObj->response_code != 200) {
+            exit();
```

Comment:
It is a bad design to exit() without any return value atleast we should return an empty object. 
The calling JS is expecting the API to return something.

Reply:
Updated in e8ef3ae